pkgs/memory_usage/lib/src/model.dart

```diff
@@ -0,0 +1,188 @@
+/// more then by [deltaMb] since previous [onUsageEvent].
+/// First [onUsageEvent] will be triggered immediately.
+class UsageEventsConfig {
+  const UsageEventsConfig(
+    this.onUsageEvent, {
+    this.deltaMb = 128,
+  });
+
+  /// A callback that is called when a snapshot is taken.
+  final UsageCallback onUsageEvent;
+
+  /// Change in memory usage to trigger [onUsageEvent].
+  final int deltaMb;
+
+  @override
+  String toString() {
+    final formatter = NumberFormat('#,###,000');
+    return 'usageEvent.deltaMb: ${formatter.format(deltaMb)}';
+  }
+}
+
+/// Configures auto-snapshotting, based on the value of `ProcessInfo.currentRss` (dart:io).
+///
+/// Automatic snapshots will begin to be taken when the rss value exceeds [thresholdMb].
+/// The snapshots will be saved to [directory].
+///
+/// The snapshots will be re-taken when the value
+/// increases more than by [increaseMb] since previous snapshot,
+/// until the size of [directory] exceeds
+/// [directorySizeLimitMb].
+///
+/// The [onSnapshot] callback is called when a snapshot is taken.
+///
+/// Set [minDelayBetweenSnapshots] to make sure snapshots do not trigger each other.
+class AutoSnapshottingConfig {
+  AutoSnapshottingConfig({
+    this.thresholdMb = 1024, // 1Gb
+    this.increaseMb = 512, // 0.5Gb
+    this.directory = 'dart_memory_snapshots',
+    this.directorySizeLimitMb = 10240, // 10Gb
+    this.minDelayBetweenSnapshots = const Duration(seconds: 10),
+    this.onSnapshot,
+  }) {
+    if (minDelayBetweenSnapshots <= Duration.zero) {
+      throw ArgumentError.value(
+        minDelayBetweenSnapshots,
+        'minDelayBetweenSnapshots',
+        'must be positive',
+      );
+    }
+  }
+
+  /// The rss value in Mb that will trigger the first snapshot.
+  final int thresholdMb;
+
+  /// The value by which the rss value should increase, since
+  /// previous snapshot, to take another snapshot.
+  ///
+  /// If [increaseMb] is null, only one snapshot will be taken.
+  final int? increaseMb;
+
+  /// The directory where snapshots will be saved.
+  ///
+  /// If the directory does not exist, it will be created.
+  ///
+  /// If the path is relative, it will be relative to the current working directory.
+  final String directory;
+
+  /// The size limit for the [directory] in Mb.
+  ///
+  /// The directory size will be checked before a snapshot is taken and saved,
+  /// so the directory may exceed the size specified by [directorySizeLimitMb]
+  /// depending on the size of the snapshot.
+  final int directorySizeLimitMb;
+
+  /// How long to wait after taking a snapshot before taking another one.
+  final Duration minDelayBetweenSnapshots;
+
+  /// A callback that is called when a snapshot is taken.
+  final SnapshotCallback? onSnapshot;
+
+  /// The absolute path to the [directory].
+  String get directoryAbsolute => path.absolute(directory);
+
+  @override
+  String toString() {
+    final formatter = NumberFormat('#,###,000');
```

Comment:
As a note to self, if the only usage of package:intl from this package is for number formatting to toString methods (I assume just used when debugging?), we should be able to prune the dep on `intl` w/o much difficulty.

Reply:
(No real feedback for this PR however as this is existing code)

Reply:
removed. thanks